lib/rubocop/cop/layout/space_inside_reference_brackets.rb

```diff
@@ -65,16 +65,24 @@ def autocorrect(node)
 
         def reference_brackets(node)
           left = left_ref_bracket(node)
-          [left, right_ref_bracket(node, left)]
+          [left, closing_bracket(node, left)]
         end
 
         def left_ref_bracket(node)
           tokens(node).reverse.find(&:left_ref_bracket?)
         end
 
-        def right_ref_bracket(node, token)
-          i = tokens(node).index(token)
-          tokens(node).slice(i..-1).find(&:right_bracket?)
+        def closing_bracket(node, opening_bracket)
+          i = tokens(node).index(opening_bracket)
+          inner_left_brackets_needing_closure = 0
+
+          tokens(node).slice(i..-1).each do |token|
```

Comment:
Wonder if there's any reason for using `#slice` over `#[]` here? 🤔

Reply:
Good call!

Reply:
Might this be a good cop candidate? We'd have to make sure the receiver is an array.